Intersect.Client/Entities/Player.cs

```diff
@@ -1534,57 +1549,104 @@ public bool TryBlock()
+                    y--;
+                    x--;
+                    break;
+
+                case (byte)Directions.UpRight:
+                    y--;
                     x++;
+                    break;
+
+                case (byte)Directions.DownLeft:
+                    y++;
+                    x--;
+                    break;
 
+                case (byte)Directions.DownRight:
+                    y++;
+                    x++;
                     break;
             }
 
             if (TryGetRealLocation(ref x, ref y, ref map))
             {
+                // Iterate through all entities
                 foreach (var en in Globals.Entities)
                 {
-                    if (en.Value == null)
+                    // Skip if the entity is not within the player's map, is null or is the current player.
+                    if (en.Value == null || en.Value == Globals.Me || en.Value.MapId != map)
                     {
                         continue;
                     }
 
-                    if (en.Value != Globals.Me)
+                    if (TargetBox.MyEntity != null)
                     {
-                        if (en.Value.MapId == map &&
-                            en.Value.X == x &&
-                            en.Value.Y == y &&
-                            en.Value.CanBeAttacked())
+                        AutoTurnToTarget(en.Value);
+                        bool canAttack = false;
+                        for (var index = AdjacentTiles.Count - 1; index >= 0; index--)
                         {
-                            //ATTACKKKKK!!!
-                            PacketSender.SendAttack(en.Key);
-                            AttackTimer = Timing.Global.Milliseconds + CalculateAttackTime();
+                            var hitBox = AdjacentTiles[index];
+                            if (hitBox[0] != en.Value.X || hitBox[1] != en.Value.Y || TargetBox.MyEntity != en.Value)
+                            {
+                                continue;
+                            }
 
-                            return true;
+                            canAttack = true;
+                            break;
                         }
+
+                        if (!canAttack)
+                        {
+                            continue;
+                        }
+
+                        // Attack the targeted entity.
+                        PacketSender.SendAttack(en.Key);
+                        AttackTimer = Timing.Global.Milliseconds + CalculateAttackTime();
+                        return true;
                     }
+
+                    if (en.Value.X != x || en.Value.Y != y)
+                    {
+                        continue;
+                    }
+
+                    // Attack the entity.
+                    PacketSender.SendAttack(en.Key);
+                    AttackTimer = Timing.Global.Milliseconds + CalculateAttackTime();
+                    return true;
```

Comment:
This is not good at all.

1. There is no check if diagonal movement is enabled. If it is not, attacks should not hit the diagonal.
  a. This introduces a new issue, how do we avoid weird conditional logic here, which I think should be extracted into an instance of a "targeting" evaluator utility class.
2. Attacks should be in a cone. With diagonal movement disabled it is just in the given direction and the original logic stands.
  a. For the cardinal directions the major axis would be strict == and the minor axis would be Abs(diff) <= 1.
  b. For the diagonals it is a little trickier, and it is X+0/1, Y+0/1 (for example in the down-right direction). Abs(diff) does not work here, but if we have another computation that can ensure we never have the "-1" value the Abs check still works, and here it actually works for both axes.

Reply:
with the recent changes i'm making : 
1) Solved ; no check is required here _but around the auto-turn to target feature_, which is already done.
2) Sorted out the second point by adding a configurable boolean that limits the ability to turn towards the opposite direction if an entity is behind the character on all it's directions.